### PR TITLE
fltk: add devel build.

### PR DESCRIPTION
**Pull request by dpo**
Closes #44806.

### CODE REVIEW

**Comment by dpo** [comment]
```
==> brew audit fltk
==> FAILED
Error: 1 problem in 1 formula
fltk:
 * devel version 1.3.x-r10866 is older than stable version 1.3.3
```

**Comment by bfontaine** [comment]
Is `1.3.x-r10866` supposed to be more recent than `1.3.3`?

**Comment by dpo** [comment]
Don't think so. That's just how the FLTK folks versioned their development snapshot.

**Comment by bfontaine** [comment]
As the audit shown we don’t allow devel versions that are older than the stable ones.

**Comment by dpo** [comment]
It's not really older. It's clearly newer because 10.11 is supported. That is the point of this PR.

**Comment by bfontaine** [comment]
Yeah, just checked upstream; the 1.3.3 version was released in November, 2014 while the 1.3.x-r10866 one was released two weeks ago.

You might use `version "1.3.3-r10866"` to ensure Hombrew think it’s a newer release.

**Comment by dpo** [comment]
`1.3.4-r10866`, no?

**Comment by bfontaine** [comment]
`1.3.4-r10866` > `1.3.4` so it won’t work when we’ll update to `1.3.4`.

File: Library/Formula/fltk.rb [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
   desc "Cross-platform C++ GUI toolkit"
   homepage "http://www.fltk.org/"
-  url "https://fossies.org/linux/misc/fltk-1.3.3-source.tar.gz"
-  sha256 "f8398d98d7221d40e77bc7b19e761adaf2f1ef8bb0c30eceb7beb4f2273d0d97"
+  revision 1
+
+  stable do
+    url "https://fossies.org/linux/misc/fltk-1.3.3-source.tar.gz"
+    sha256 "f8398d98d7221d40e77bc7b19e761adaf2f1ef8bb0c30eceb7beb4f2273d0d97"
+    depends_on MaximumMacOSRequirement => :yosemite
```
Comment (bfontaine) [inline Library/Formula/fltk.rb]: Please keep the original comment above this line:
`# Fltk 1.3.4 include support for El Capitan. Remove on update.`

File: Library/Formula/fltk.rb [hunk 2]
```diff
@@ -14,20 +32,15 @@ class Fltk < Formula
...
   end
 
-  # Fixes issue with -lpng not found.
-  # Based on: https://trac.macports.org/browser/trunk/dports/aqua/fltk/files/patch-src-Makefile.diff
-  patch :DATA
-
   def install
     ENV.universal_binary if build.universal?
+
+    system "NOCONFIGURE=1 ./autogen.sh" if build.devel?
```
Comment (bfontaine) [inline Library/Formula/fltk.rb]: Please export the env variable separatly:
```rb
if build.devel?
  ENV["NOCONFIGURE"] = "1"
  system "./autogen.sh"
end
```

Comment (dpo) [inline Library/Formula/fltk.rb]: That's 4 lines instead of 1. Why is it necessary?

Comment (bfontaine) [inline Library/Formula/fltk.rb]: `system "foo bar"` runs in a shell while `system "foo", "bar"` doesn’t. We thus prefer the later.

**Comment by dpo** [comment]
Done.

File: Library/Formula/fltk.rb [hunk 3]
```diff
@@ -1,8 +1,28 @@
...
+
+    # Fixes issue with -lpng not found.
+    # Based on: https://trac.macports.org/browser/trunk/dports/aqua/fltk/files/patch-src-Makefile.diff
+    patch :DATA
+  end
+
+  devel do
+    url "http://fltk.org/pub/fltk/snapshots/fltk-1.3.x-r10866.tar.gz"
+    sha256 "4fd4911a1da99c2fa1e6cc0c985b3b8645a7e954802230338fb513ae880ff2cc"
+    version "1.3.3-r10866" # convince brew that this is older than stable
```
Comment (bfontaine) [inline Library/Formula/fltk.rb]: This is *newer* than stable :wink:

File: Library/Formula/fltk.rb [hunk 4]
```diff
@@ -1,8 +1,28 @@
 class Fltk < Formula
   desc "Cross-platform C++ GUI toolkit"
   homepage "http://www.fltk.org/"
-  url "https://fossies.org/linux/misc/fltk-1.3.3-source.tar.gz"
-  sha256 "f8398d98d7221d40e77bc7b19e761adaf2f1ef8bb0c30eceb7beb4f2273d0d97"
+  revision 1
```
Comment (bfontaine) [inline Library/Formula/fltk.rb]: You don’t need a revision since you didn’t modify the stable version.

**Comment by dpo** [comment]
Sorry ;-). This is the PR that never ends.

File: Library/Formula/fltk.rb [hunk 5]
```diff
@@ -14,20 +33,18 @@ class Fltk < Formula
...
 
-  # Fixes issue with -lpng not found.
-  # Based on: https://trac.macports.org/browser/trunk/dports/aqua/fltk/files/patch-src-Makefile.diff
-  patch :DATA
-
   def install
     ENV.universal_binary if build.universal?
+
+    if build.devel?
+      ENV["NOCONFIGURE"] = 1
```
Comment (bfontaine) [inline Library/Formula/fltk.rb]: Use `"1"` instead of `1` here. The build fails otherwise.

**Comment by dpo** [comment]
It was so nice and easy with the one-liner ;-).

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by dpo** [comment]
All good now. Thanks!

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/004f13435c91c96ab512e226dfa4783c14e727ea. Thank you for pushing through with this @dpo :smiley_cat:.

**Comment by dpo** [comment]
Thank you.

**Pull request closed without merging**